### DISCUSSION TITLE
Providing effects and store for feature to standalone app without routing (how to create an EnvironmentInjector?)

### DISCUSSION THREAD

**Discussion by alfmos** [ngrx/platform#4203] [Q&A]
We have an Angular app that we are converting to standalone. After some trial and error everything went fine, except for a single case.
We have a component that asks the user for some data confirmation. This happens usually every two years. Clearly we want to load this component only when needed. 
The component itself is loaded inside a dialog, but it has some state and some effects. Now, to provide state and effects I need an **EnvironmentInjector** (as ProvideEffects returns an EnvironmentProvider). 
I can't use providers array in bootstrapApplication as it is global, and I haven't a route, as the component is loaded inside a dialog, no matter what is the page selected.
the angular guide [(https://angular.io/guide/standalone-components#environment-injectors)](https://angular.io/guide/standalone-components#environment-injectors) says
![immagine](https://github.com/ngrx/platform/assets/46923488/0bd8679a-72f7-4604-8aa0-71ccb48633f8)

So I think the only option left is to create an EnvironmentInjector with CreateEnvironmentInjector. This, from what I have understood, would imply injecting an ApplicationRef to get the root injector and pass it as parent to CreateEnvironmentInjector.
I don't know if this is possible with provideEffects, or if there is a canonical way to do this, or simply a better way.
Could someone point me in the right direction?

**Comment by TitouanLarnicol** [comment]
I'm starting this thread again because I have the same problem and I can't find many resources on the subject.
Do we have any more information on how to solve or get around this problem ?

**Comment by brandonroberts** [comment]
Yes, its possible to do. `createEnvironmentInjector` accepts a `providers` array, in addition to a parent `EnvironmentInjector`. You can create the environment injector manually in your component class, and put the state/effects providers there.

Example

```ts
createEnvinronmentInjector([
  provideState(),
  provideEffects()
], parentInjector);
```
